### PR TITLE
[WIP] Members magic link

### PR DESCRIPTION
**Pull request by allouis**
Updates for members to support signin with magic link.

Includes migrations and refactors

Still waiting on version bumps

### CODE REVIEW

**Comment by allouis** [comment]
@rishabhgrg and @gargol This includes a migration, could I get some :eyes: on that please?

File: core/server/api/canary/members.js [hunk 1]
```diff
@@ -28,8 +29,14 @@ const members = {
         ],
         validation: {},
```
Comment (gargol) [inline core/server/api/canary/members.js]: @allouis is there a reason why there's no validation at all? thought at least email could go through a basic check like we do for invitations - https://github.com/TryGhost/Ghost/blob/7b761a8/core/server/api/canary/utils/validators/input/invitations.js#L34

**Review by naz — APPROVED** [review]
Migrations worked well. Just one little comment around validations but don't think it's a blocker :+1:

**Review by rshbhgrg — APPROVED** [review]

File: core/server/api/canary/members.js (hunk 1 above)
Comment (allouis) [reply core/server/api/canary/members.js]: The idea is that this controller is just proxying to a service, so unsure whether that validation should be handled in the service! Gonna leave it out of this PR for now, but something we can look at during cleanup

**Pull request merged**